src/shared/Atlassian.Bitbucket.UI/Controls/TesterWindow.axaml.cs

```diff
@@ -0,0 +1,68 @@
+using Atlassian.Bitbucket.UI.ViewModels;
+using Atlassian.Bitbucket.UI.Views;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Interactivity;
+using Avalonia.Markup.Xaml;
+using Microsoft.Git.CredentialManager;
+using Microsoft.Git.CredentialManager.Interop.Linux;
+using Microsoft.Git.CredentialManager.Interop.MacOS;
+using Microsoft.Git.CredentialManager.Interop.Posix;
+using Microsoft.Git.CredentialManager.Interop.Windows;
+using Microsoft.Git.CredentialManager.UI.Controls;
+
+namespace Atlassian.Bitbucket.UI.Controls
+{
+    public class TesterWindow : Window
+    {
+        private readonly IEnvironment _environment;
+
+        public TesterWindow()
+        {
+            InitializeComponent();
+#if DEBUG
+            this.AttachDevTools();
+#endif
+
+            if (PlatformUtils.IsWindows())
+            {
+                _environment = new WindowsEnvironment(new WindowsFileSystem());
+            }
+            else
+            {
+                IFileSystem fs;
+                if (PlatformUtils.IsMacOS())
+                {
+                    fs = new MacOSFileSystem();
+                }
+                else
+                {
+                    fs = new LinuxFileSystem();
+                }
+
+                _environment = new PosixEnvironment(fs);
+            }
```

Comment:
I haven't gotten through the whole review yet, but this kind of pattern makes me think we could use a base class, and have platform specific implementations and isolate the platform switch to a single factory style function, or a DI container registration, etc.

Reply:
Yeah the 'factory' is normally in the `CommandContext` constructor. The `CommandContext` creates and holds references to all these components. We're only manually new-ing the select few components here in the UI tester, which is not part of normal code execution.